Community/Tdarr_Plugin_bsh1_Boosh_FFMPEG_QSV_HEVC.js

```diff
@@ -0,0 +1,400 @@
+          so bitrate_cutoff is multiplied by 2! Cutoff now temporarily ${inflatedCutoff}. 
+          \n The file is NOT above this new cutoff. Exiting plugin. \n\n`;
+          return response;
+        }
+      }
+    }
+
+    // Are we encoding 10 bit files?
+    if (inputs.enable_10bit === 'true') {
+      // Check if video stream is 10bit. Currently set profile main10 and pixel format.
+      // Still need to test if this really works correctly
+      if (file.ffProbeData.streams[i].profile === 'High 10'
+        || file.ffProbeData.streams[i].bits_per_raw_sample === '10') {
+        main10 = '-profile:v main10 -pix_fmt p010le';
+        response.infoLog += '☑ File is 10bit or HDR. Setting Main10 Profile & pixel format \n\n';
+      }
+    }
+
+    // Increment videoIdx.
+    videoIdx += 1;
+  }
+
+  // Set bitrateSettings variable using bitrate information calulcated earlier.
+  bitrateSettings = `-b:v ${targetBitrate}k -minrate ${minimumBitrate}k `
+    + `-maxrate ${maximumBitrate}k -bufsize ${currentBitrate}k`;
+  // Print to infoLog information around file & bitrate settings.
+  response.infoLog += `\nContainer for output selected as ${inputs.container}. \n`;
+  response.infoLog += `The current file bitrate = ${currentBitrate} \n`;
+  response.infoLog += 'Bitrate settings: \n';
+  response.infoLog += `Target = ${targetBitrate} \n`;
+  response.infoLog += `Minimum = ${minimumBitrate} \n`;
+  response.infoLog += `Maximum = ${maximumBitrate} \n`;
+
+  // Codec will be checked so it can be transcoded correctly
+  // (QSV doesn't support HW decode of all older codecs, h263 & mpeg1 are SW based currently)
+  // If 10 bit is being used then don't hardware decode, this can cause issues.
+  // Best just to cpu decode to ensure it works.
+  if (main10 === '') {
+    if (file.video_codec_name === 'h263') {
+      response.preset = '-hwaccel qsv -c:v h263';
+    } else if (file.video_codec_name === 'h264') {
+      response.preset = '-hwaccel qsv -c:v h264_qsv';
+    } else if (file.video_codec_name === 'mjpeg') {
+      response.preset = '-hwaccel qsv -c:v mjpeg_qsv';
+    } else if (file.video_codec_name === 'hevc') {
+      response.preset = '-hwaccel qsv -c:v hevc_qsv';
+    } else if (file.video_codec_name === 'mpeg1') {
+      response.preset = '-hwaccel qsv -c:v mpeg1';
+    } else if (file.video_codec_name === 'mpeg2') {
+      response.preset = '-hwaccel qsv -c:v mpeg2_qsv';
+    } else if (file.video_codec_name === 'vc1') {
+      response.preset = '-hwaccel qsv -c:v vc1_qsv';
+    } else if (file.video_codec_name === 'vp8') {
+      response.preset = '-hwaccel qsv -c:v vp8_qsv';
+    } else if (file.video_codec_name === 'av1') {
+      response.preset = '-hwaccel qsv -c:v av1';
+    }
+  }
+
+  response.preset += `,-map 0 -c:v hevc_qsv ${bitrateSettings} `
+    + `-preset ${inputs.encoder_speedpreset} -look_ahead 1 ${main10} 
+    -c:a copy -c:s copy -max_muxing_queue_size 9999 ${extraArguments}`;
```

Comment:
Plugin will fail if there's no audio or subtitles.

Reply:
This is not what I've experienced. I've put through video only files and the encode completed fine with no error. I believe this cmd argument will copy audio and subtitles if they are found, but won't error if none are present.